scripts/sync-exported-npm-with-cli.js

```diff
@@ -0,0 +1,78 @@
+ * However, the Cypress binary will also ship an export for `cypress/react` that's guaranteed to work
+ * with this version of the binary
+ */
+const shell = require('shelljs')
+const path = require('path')
+const packlist = require('npm-packlist')
+const fs = require('fs')
+
+shell.set('-v') // verbose
+shell.set('-e') // any error is fatal
+
+// This script will be run in a postbuild task for each npm package
+// that will be re-exported by Cypress
+const currentPackageDir = process.cwd()
+
+// 1. We'll run npm's own "packlist" against the npm package to be published (@cypress/react, etc)
+// to make sure we don't miss any files when we copy them over to the CLI package
+// The files that will be returned here are the ones from @cypress/react's package.json['files'] key.
+packlist({ path: currentPackageDir })
+.then((files) => {
+  // 2. Move all of the files that would be published under @cypress/react
+  // to be copied under cli/react (drop the @cypress namespace)
+  const cliPath = path.join(__dirname, '..', 'cli')
+
+  // Typically, these packages are independently published as @cypress/package-name
+  // e.g. @cypress/vue => import whatever from 'cypress/vue'
+  // The files will wind up at cypress/cli/vue/*
+  const currentPackageConfig = require(path.join(process.cwd(), 'package.json'))
+  const exportName = currentPackageConfig.name.replace('@cypress/', '')
+  const outDir = path.join(cliPath, exportName)
+
+  // 3. For each file, mkdir if not exists, and then copy the dist'd assets over
+  // Shell is synchronous by default, but we don't actually need to await for the results
+  // to write to the `cliPackageConfig` at the end
+  files.forEach((f) => {
+    // mkdir if not exists
+    const { dir } = path.parse(f)
+
+    if (dir) {
+      shell.mkdir('-p', path.join(outDir, dir))
+    }
+
+    shell.cp(path.join(currentPackageDir, f), path.join(outDir, f))
+  })
+
+  // After everything is copied, let's update the Cypress cli package.json['exports'] map.
+  const isModule = currentPackageConfig.type === 'module'
+
+  const cliPackageConfig = require(path.join(cliPath, 'package.json'))
+
+  const subPackageExports = cliPackageConfig.exports[`./${exportName}`] = {}
+  const esmEntry = isModule ? currentPackageConfig.main : currentPackageConfig.module
+
+  if (esmEntry) {
+    // ./react/dist/cypress-react-esm.js, etc
+    subPackageExports.import = `./${exportName}/${esmEntry}`
+  }
+
+  if (!isModule) {
+    // ./react/dist/cypress-react-cjs.js, etc
+    subPackageExports.require = `./${exportName}/${currentPackageConfig.main}`
+  }
+
+  if (!cliPackageConfig.files.includes(exportName)) {
+    cliPackageConfig.files.push(exportName)
+  }
+
+  const output = `${JSON.stringify(cliPackageConfig, null, 2) }\n`
+
+  // eslint-disable-next-line no-console
+  console.log('Writing to CLI package.json for', exportName)
+
+  fs.writeFileSync(path.join(cliPath, 'package.json'), output, 'utf-8')
+})
```

Comment:
The tiniest thing but this seems like pretty important logic to test - do we need some kind of basic test to ensure that the `package.json` updates correctly, and that `import { mount } from 'cypress/vue` is actually working? We don't have any coverage for this right now, from what I can see.

I wonder if we can update all internal code to use the deep import syntax in the future. Is this something we plan to do?

Reply:
We can test it by adding code here https://github.com/cypress-io/cypress-example-kitchensink/pull/528

I'll do that. It'll break all other pipelines until this is merged, maybe I'll create a new branch for it (in kitchensink).

Reply:
I was also wondering if we can convert a couple of the existing system tests to `import { mount } from 'cypress/vue'` rather than `import { mount } from '@cypress/vue'`.

Reply:
Oh, here https://github.com/cypress-io/cypress-example-kitchensink/pull/547/files

Reply:
That probably only works in `test-binary`, where we have tests that run in a docker container against the compiled binary. But still worth looking into. Just an extremely simple 'can import' test there.

Reply:
Yeah in general I feel uncomfortable that we don't have a lot of post-build infrastructure for testing the output of Circle CI's build scripts. The kitchen sink is the right place. I put a PR up above that will at least fail if those packages aren't declared in the exports map.